tests/hazmat/asn1/test_serialization.py

```diff
@@ -23,35 +36,28 @@ class TestEncodeInteger:
             (-129, b"\x02\x02\xff\x7f"),
         ],
     )
-    def test_encode_int(self, value: int, expected: bytes) -> None:
-        encoded = asn1.encode_der(value)
-        assert encoded == expected, (
-            f"Failed for {value}: got {encoded.hex()}, expected"
-            f"{expected.hex()}"
-        )
+    def test_int(self, obj: int, obj_bytes: bytes) -> None:
```

Comment:
Instead of using `pytest.mark.parametrize`, can we just have `assert_roundtrips` take a list of test cases?